### PR TITLE
Feature/multithreaded miner btc 675

### PR DESCRIPTION
**Pull request by blackyblack**

### CODE REVIEW

File: include/veriblock/blockchain/miner.hpp [hunk 1]
```diff
@@ -20,19 +18,88 @@ template <typename Block, typename ChainParams>
...
-      // to guarantee that miner will not create exactly same blocks even if
-      // time and merkle roots are equal for prev block and new block
-      block.setNonce(nonce++);
-      if (block.getNonce() >=
-          (std::numeric_limits<decltype(block.getNonce())>::max)()) {
-        block.setTimestamp(block.getTimestamp() + 1);
-        nonce = 0;
+    std::mutex lock;
+    std::condition_variable finishedSignal;
+    auto workers = createWorkers(block, finishedSignal, lock);
```
Comment (Warchant) [inline include/veriblock/blockchain/miner.hpp:80]: can we create workers once?

Comment (blackyblack) [reply include/veriblock/blockchain/miner.hpp:80]: yes but a bit annoying to have worker initalization in separate parts of code.

Comment (Warchant) [reply include/veriblock/blockchain/miner.hpp:80]: Maybe, create them when Miner is created?

Comment (Warchant) [reply include/veriblock/blockchain/miner.hpp:80]: Imagine mining 100k BTC blocks with this code

**Comment by Warchant** [comment]
Obsolete

**Pull request closed without merging**